src/pages/Navgurukul/merakiAdmission/index.js

```diff
@@ -0,0 +1,211 @@
+        middleName: userDetails.middleName,
+        lastName: userDetails.lastName,
+        mobileNumber: userDetails.mobileNumber,
+      };
+      const queryString = Object.keys(params)
+        .map((filter) => {
+          if (params[filter]) {
+            return `${filter}=${params[filter]}`;
+          }
+          return null;
+        })
+        .filter((item) => item)
+        .join("&");
+
+      const url = `${testUrl}${enrolmentKey}?${queryString}`;
+
+      window.open(url, "_blank");
+      return response;
+    } catch (e) {}
+  };
+
+  const giveTest = async () => {
+    if (
+      !userDetails.firstName ||
+      !userDetails.lastName ||
+      !userDetails.mobileNumber
+    ) {
+      toast.error(
+        "To attempt the test, it is compulsory to enter your First Name, Last Name and Mobile Number. Middle Name is optional, you can choose not to enter.",
+        {
+          position: toast.POSITION.BOTTOM_RIGHT,
+        }
+      );
+      return;
+    }
+    if (userDetails.mobileNumber.toString().length !== 10) {
+      toast.error("Please give 10 digits of the mobile number.", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+      });
+      return;
+    }
+    await isDuplicate();
+  };
+
+  const isDuplicate = () => {
+    const { firstName, middleName, lastName, mobileNumber } = userDetails;
+    axios
+      .get(`${process.env.REACT_APP_CHANAKYA_BASE_URL}/check_duplicate`, {
+        params: {
+          Name: firstName.concat(" ", middleName, lastName),
+          Number: mobileNumber,
+        },
+      })
+      .then(async (data) => {
+        const response = data.data.data;
+        const stage = response.pendingInterviewStage;
+        const url = `https://admissions.navgurukul.org/check_duplicate/Name=${firstName}${middleName}${lastName}&Number=${mobileNumber}&Stage=${stage}`;
+        if (response.alreadyGivenTest) {
+          window.open(url, "_blank");
+        } else {
+          generateTestLink();
+        }
+      });
+  };
+
+  const changeHandler = (e) => {
+    setUserDetails({ ...userDetails, [e.target.name]: e.target.value });
+  };
+
+  return (
+    <div className="admission">
+      <div className="left-section">
+        <YouTube className={"video-alumni"} videoId={`vuSwndj5cbs`} />;
+        <div className="video-label">Experience of NG Alumni & Graduates</div>
+      </div>
+      <div className="right-section">
+        <div className="test-form">
+          <h2 className="scholarship-test">
+            Software Engineering Scholarship Test
+          </h2>
+          <div className="input-form">
+            <div className="form-row">
+              <div className="input-field-test">
+                <span>First Name</span>
+                <input
+                  type="text"
+                  placeholder="Abhi..."
+                  value={userDetails.firstName}
+                  name="firstName"
+                  onChange={changeHandler}
```

Comment:
It may be good to add the [`required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required) attribute here and on the Last Name, and Mobile Number text inputs.
```suggestion
                  required
                  onChange={changeHandler}
```